salt/modules/file.py

```diff
@@ -2171,14 +2171,14 @@ def replace(path,
         if not_found_content is None:
             not_found_content = repl
         if prepend_if_not_found:
-            new_file.insert(0, not_found_content + b'\n')
+            new_file.insert(0, not_found_content + os.linesep)
```

Comment:
This is going to break on Python 3, `os.linesep` is not a bytestring.

```
Python 3.6.2 (default, Jul 20 2017, 03:52:27)                                      
[GCC 7.1.1 20170630] on linux            
Type "help", "copyright", "credits" or "license" for more information.             
>>> import os       
>>> b'foo' + os.linesep                  
Traceback (most recent call last):       
  File "<stdin>", line 1, in <module>    
TypeError: can't concat str to bytes     
>>> 
```

You should be using `salt.utils.to_bytes(os.linesep)`.